secboot/secboot_sb.go

```diff
@@ -178,11 +186,38 @@ func UnlockVolumeUsingSealedKeyIfEncrypted(disk disks.Disk, name string, sealedE
 	return res, nil
 }
 
-// UnlockEncryptedVolumeUsingKey unlocks an existing volume using the provided key.
-func UnlockEncryptedVolumeUsingKey(disk disks.Disk, name string, key []byte) (UnlockResult, error) {
+func deviceHasPlainKey(device string) (bool, error) {
+	slots, err := sbListLUKS2ContainerUnlockKeyNames(device)
```

Comment:
if we need to do this kind of checks often we might want to have a central object so we can cache things during bootstrap

Reply:
We need this check only once per boot.